docs/migrating-to-tool-kit.md

```diff
@@ -0,0 +1,181 @@
+hook. A common example would be the `@dotcom-tool-kit/eslint` and
+`@dotcom-tool-kit/mocha` plugins, which both define tasks that use the
+`test:local` hook. Tool Kit does not assume the order you want those to run, as
+in some cases one of the tasks might depend on the other, so you need to declare
+the order explicitly. This is expanded on in its own [documentation
+page](./resolving-hook-conflicts.md), but the migration tool takes you through
+the process. You choose each task in the order you want them to run in. If you
+don't want some of the tasks to be included in the hook at all, just select the
+`finish` option after the rest of the tasks have been selected and the remainder
+won't be run.
+
+### Setting Options
+
+```
+Please now configure the options for the ${plugin} plugin.
+Set a value for '${option}'
+```
+
+Some plugins require configuration to work, whereas others might have optional
+settings that you may want to tweak. You must set all the required options to
+continue, though naturally the optional ones may be skipped. In the future, we
+will add support for default values in the migration tool (they are already
+available within Tool Kit itself,) so that for most cases you could accept
+default values, even for fields that are required.
+
+### Migrating Your Makefile
+
+```
+We recommend deleting your old Makefile as it will no longer be used. In the
+future you can run tasks with 'npm run' instead. Make sure that you won't be
+deleting any task logic that hasn't already been migrated to Tool Kit. If you
+find anything that can't be handled by Tool Kit then please let the Platforms
+team know.
+
+We've found some targets in your Makefile which could be migrated to Tool Kit:
+- Your ${makefile target} target is likely handled by the ${target} hook in Tool Kit
+
+We don't know if these other Makefile targets can be migrated to Tool Kit.
+Please check what they're doing:
+- ${makefile target}
+- ${makefile target}
+```
+
+Finally, the tool will give guidance on what the do with the Makefile in your
+project that was integrated with n-gage. It will try and recommend equivalent
+hooks based on the name of the targets within the Makefile (the `test:local`
+hook for a `test` target, for example,) but sometimes you'll want to leave some
+of the targets in at the beginning of the migration and gradually transition
+everything over to Tool Kit. Regardless, this step just provides some
+suggestions and the migration tool doesn't perform any actions so you'll need to
+delete the Makefile yourself if you don't think it's needed anymore. Remember
+that – assuming you didn't delete it in the earlier step – some of the `make`
+commands come from `n-gage` itself and won't necessarily be in the Makefile,
+such as `make install`.
+
+## Migrating Your Heroku Pipeline
+
+One thing that the migration tool cannot automate is reconfiguring your Heroku
+pipeline to integrate properly with Tool Kit. You will need to ensure that your
```

Comment:
**suggestion:** perhaps we can add here, that

On top of installing the plugin `circleci-heroku`, which provides hooks that circleci runs such as 'deploy:review' that creates a review app,  you'll have to ensure that your
pipeline is connected to the project's ...

just to be clear that you'll still need the heroku plugin as well as following the step by step guide

Reply:
The `front-end` plugin will install `circleci-heroku` automatically so I'm not sure if we need to be explicit about it?

Reply:
ah right of course, I agree - we don't have to be explicit about it then